react-hot-button-component/src/hot-button.jsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+
+let clicks = 0;
+export default class HotButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+    this.state = { clickCounter: clicks };
+
+  }
+
+  handleClick() {
+    this.setState({ clickCounter: clicks++ });
```

Comment:
Here you should use state to increment itself, using an out side variable could lead to unpredictable results.

```JAVASCRIPT
this.setState({ clickCounter: this.state.clickCounter + 1 });
```